samza-api/src/main/java/org/apache/samza/operators/functions/FilterFunction.java

```diff
@@ -22,7 +22,8 @@
 
 
 /**
- * A function that specifies whether a message should be retained for further processing or filtered out.
+ * Specifies whether a message should be retained for further processing.
```

Comment:
Can a java Predicate impl be used interchangeably with this interface? If so, that'd be pretty powerful and worth mentioning. 

Top of my head, we'd need this interface to extend Predicate, so the answer is probably no. Just throwing it out there.